# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GCInternals.h
# Commit: f78d139e1d0e
# Full Hash: f78d139e1d0e7d1760239cfbe4b8f67e75e1d29c
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-09 21:37:17
# Regressor Bug: 1643586
# File Overlap Count: 1
# Description:
#   Bug 1643586 - Part 4: Remove flag for disabling compartment checks from JSContext r=jandem
#   
#   This is only used in one place and it's simpler to check the tracer kind there.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D140191
# ==============================================================================

diff -r 8611c3ef5a77 -r f78d139e1d0e js/src/gc/GCInternals.h
--- a/js/src/gc/GCInternals.h	Wed Mar 09 10:17:07 2022 +0000
+++ b/js/src/gc/GCInternals.h	Wed Mar 09 10:17:07 2022 +0000
@@ -231,25 +231,6 @@
 };
 #endif /* JS_GC_ZEAL */
 
-class MOZ_RAII AutoDisableCompartmentCheckTracer {
-#ifdef DEBUG
-  JSContext* cx_;
-  bool prev_;
-
- public:
-  AutoDisableCompartmentCheckTracer()
-      : cx_(TlsContext.get()), prev_(cx_->disableCompartmentCheckTracer) {
-    cx_->disableCompartmentCheckTracer = true;
-  }
-  ~AutoDisableCompartmentCheckTracer() {
-    cx_->disableCompartmentCheckTracer = prev_;
-  }
-#else
- public:
-  AutoDisableCompartmentCheckTracer(){};
-#endif
-};
-
 class MOZ_RAII AutoPoisonFreedJitCode {
   JSFreeOp* const fop;
 